mixpanel/tests/test_tasks.py

```diff
@@ -47,19 +47,40 @@ def set_mp_settings(self):
 
     def patch_network(self):
         self.old_get_connection = EventTracker._get_connection
-        self.conn = Mock()
+
+        class Response(object):
+            status = 200
+            reason = 'OK'
+
+            def read(*args, **kwargs):
+                return '1'
+        response = Response()
+        self.response = response
+
+        class Connection(object):
+            def __init__(self, *args, **kwargs):
+                self._request_call_args = []
+
+            def getresponse(self, *args, **kwargs):
+                return response
+
+            def request(self, *args, **kwargs):
+                self._request_call_args.append(args)
+
+            @property
+            def request_call_args(self):
+                return self._request_call_args
+
+            def close(self, *args, **kwargs):
+                pass
         EventTracker._get_connection = lambda task: self.conn
-        self.response = Mock()
```

Comment:
I couldn't figure out how to get this working with mock. So I switched to something mock like. And updated everything to work with that.

To note, tests on master were failing for me. So it's not like I could jump back to debug.